src/main/io/dataEdition.c

```diff
@@ -0,0 +1,1359 @@
+static float valueFloat = 0;
+
+#if (DATA_EDITION_FULL_VERSION)
+extern uint16_t rssi;
+static uint32_t lastAlarmSoundTime;
+static bool isAlarmSoundOn 		= true;
+extern uint8_t numberOfBaudrateValues ;
+extern uint8_t motorCount;
+extern uint8_t servoCount;
+#endif
+
+//======================================  Utilities  =======================================
+void initialiseOledDisplayPage(void)
+{
+    memset(&structureOfPage, 0, sizeof(structureOfPage));
+}
+
+void initDataEditionStructures(void)    // called in main
+{
+#ifdef DISPLAY
+    if (featureConfigured(FEATURE_DISPLAY)) {
+    initialiseOledDisplayPage();
+    }
+#endif
+	onGoingDataEditionWithSticks = false;
+}
+
+static void initializeEditionModeSpace(void)
+{
+	for (int i = 0; i < DEFAULT_DISPLAY_LINES; i++) {
+		structureOfPage.displayLine[i].isActive = false;
+		structureOfPage.displayLine[i].itemsOnLine = 1;
+	}
+	for (int i = 0; i < MAX_OF_EDITABLE_ITEMS; i++)
+		structureOfPage.editField[i].alive = false;
+
+	selectedLine = 1;
+	editionLine = -1;
+    selectedItem = -1;
+    shouldInitialiseThePage = false;
+	onGoingEditionProcess = false;
+	numberOfSecondaryPagesForThatPage = 0;
+}
+
+static uint8_t firstItemOnLine(uint8_t line)
+{
+    return line * THREE_ITEMS_PER_LINE;
+}
+
+static void highlightSelectedItemOnOledDisplay(void)
+{
+	switch (structureOfPage.editField[selectedItem].color){
+		case PIXEL_MIXED :
+			oledCharFormat = INVERSE_CHAR_FORMAT;
+			i2c_OLED_set_xy(positionInLineOfInvertedChar, editionLine);
+			i2c_OLED_send_char(contentOfPage.lineOfText[editionLine][positionInLineOfInvertedChar]);
+			break;
+		case PIXEL_INVERTED :
+			if (structureOfPage.editField[selectedItem].color == PIXEL_INVERTED) {
+				oledCharFormat = INVERSE_CHAR_FORMAT;
+				for(int i = structureOfPage.editField[selectedItem].leftPosition; i <= structureOfPage.editField[selectedItem].rightPosition; i++){
+					i2c_OLED_set_xy(i, editionLine);
+					i2c_OLED_send_char(contentOfPage.lineOfText[editionLine][i]);
+				}
+			}
+			break;
+	}
+	oledCharFormat = NORMAL_CHAR_FORMAT;
+}
+
+static void completeStringWithSpaces(uint8_t lineIndex)
+{
+    uint8_t length = strlen(contentOfPage.lineOfText[lineIndex]);
+
+    while (length < sizeof(contentOfPage.lineOfText[lineIndex]))
+    	contentOfPage.lineOfText[lineIndex][length++] = 0x20;
+}
+
+
+#if (DATA_EDITION_FULL_VERSION)
+static bool shouldTriggerAlarmNow(void)
+{
+	return ((millis() - lastAlarmSoundTime) >= (uint32_t)(masterConfig.telemetryConfig.hottAlarmSoundInterval * MILLISECONDS_IN_A_SECOND));
+}
+
+static bool isBatteryLevelOK(void)
+{
+	if ((featureConfigured(FEATURE_VBAT)) && (getBatteryState() != BATTERY_OK))
+		return false;
+	return true;
+}
+
+static bool isRssiLevelOK(void)
+{
+	if (((masterConfig.rxConfig.rssi_channel > 0) || (featureConfigured(FEATURE_RSSI_ADC))) && (rssi <= masterConfig.rxConfig.rssi_warningLevel))
+	    return false;
+	return true;
+}
+
+static void checkForBatteryAndRSSIWarningLevels(void)
```

Comment:
ooops, just a residue, should have been removed from previous version

Le 05/01/2016 13:32, Petr Ledvina a écrit :
>
> In src/main/io/dataEdition.c 
> <https://github.com/cleanflight/cleanflight/pull/1654#discussion_r48840234>:
>
> > +
> > +static bool isBatteryLevelOK(void)
> > +{
> > +	if ((featureConfigured(FEATURE_VBAT)) && (getBatteryState() != BATTERY_OK))
> > +		return false;
> > +	return true;
> > +}
> > +
> > +static bool isRssiLevelOK(void)
> > +{
> > +	if (((masterConfig.rxConfig.rssi_channel > 0) || (featureConfigured(FEATURE_RSSI_ADC))) && (rssi <= masterConfig.rxConfig.rssi_warningLevel))
> > +	    return false;
> > +	return true;
> > +}
> > +
> > +static void checkForBatteryAndRSSIWarningLevels(void)
>
> This code does nothing?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/cleanflight/cleanflight/pull/1654/files#r48840234>.
>